### DISCUSSION TITLE
[Speech To Text]The problem of speech recognition is unavailable

### DISCUSSION THREAD

**Discussion by QuicklyPeople** [CommunityToolkit/Maui#1380] [Q&A] [answered]
I try three phones of android  at least which version is 12 and13 and all of them has the same problem that speech recognition  is not available.
I have no idea how to fit it.By the way,the permission of request is normal and the code is same to the communitytoolkit.
Here are the codes.

```c#
private async void Listen()
  {
      var isGranted = await speechToText.RequestPermissions(cancellationToken);
      if (!isGranted)
      {
          await Toast.Make("Permission not granted").Show(CancellationToken.None);
          return;
      }

      var recognitionResult = await speechToText.ListenAsync(
                                    CultureInfo.GetCultureInfo("en-US"),
                                    new Progress<string>(partialText =>
                                    {
                                        RecognitionText += partialText + " ";
                                    }), cancellationToken);


      if (recognitionResult.IsSuccessful)
      {
          RecognitionText = recognitionResult.Text;
      }
      else
      {
          await Toast.Make(recognitionResult.Exception?.Message ?? "Unable to recognize speech").Show(CancellationToken.None);
      }
  }
``` 
![recognitionResult](https://github.com/CommunityToolkit/Maui/assets/141700313/158fd85d-395d-476f-9d12-99763e63b1e9)

  {SpeechToTextResult { IsSuccessful = False, Text = , Exception = Microsoft.Maui.ApplicationModel.FeatureNotSupportedException: Speech Recognition is not available on this device
   at CommunityToolkit.Maui.Media.SpeechToTextImplementation.InternalListenAsync(CultureInfo culture, IProgress`1 recognitionResult, CancellationToken cancellationToken) in /_/src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.android.cs:line 34
   at CommunityToolkit.Maui.Media.SpeechToTextImplementation.ListenAsync(CultureInfo culture, IProgress`1 recognitionResult, CancellationToken cancellationToken) in /_/src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.shared.cs:line 21 }}

**Comment by bijington** [comment, marked as answer]
The exception message indicates that speech to text isn't supported on the device. Can you confirm whether there are hardware specifics required to support speech to text?

**Reply by QuicklyPeople** [reply]
Oh，it is so embarrassed my phone maybe not support speech to text which need the Google Search app for the service to work...

**Comment by CliffAgius** [comment]
Closed as answered
